internal/envoy/v3/listener.go

```diff
@@ -241,6 +243,20 @@ func (b *httpConnectionManagerBuilder) MergeSlashes(enabled bool) *httpConnectio
 	return b
 }
 
+func (b *httpConnectionManagerBuilder) ServerHeaderTransformation(value contour_api_v1alpha1.ServerHeaderTransformationType) *httpConnectionManagerBuilder {
+	switch value {
+	case contour_api_v1alpha1.OverwriteServerHeader:
+		b.serverHeaderTransformation = http.HttpConnectionManager_OVERWRITE
+	case contour_api_v1alpha1.AppendIfAbsentServerHeader:
+		b.serverHeaderTransformation = http.HttpConnectionManager_APPEND_IF_ABSENT
+	case contour_api_v1alpha1.PassThroughServerHeader:
+		b.serverHeaderTransformation = http.HttpConnectionManager_PASS_THROUGH
+	default:
+		return nil
```

Comment:
You can drop the default case here, it's causing panics when no server header transformation is configured. We want to return `b` here which is already being done below.